django_ipam/base/models.py

```diff
@@ -76,3 +79,29 @@ def clean(self):
         for ip in swapper.load_model("django_ipam", "IpAddress").objects.filter().values():
             if self.id != ip["id"] and ip_address(self.ip_address) == ip_address(ip["ip_address"]):
                 raise ValidationError({'ip_address': _('IP address already used.')})
+
+    def export_csv(self, subnet_id, writer, queryset):
+        IpAddress = swapper.load_model("django_ipam", "IpAddress")
+        subnet = swapper.load_model("django_ipam", "Subnet").objects.get(pk=subnet_id)
+        writer.writerow([subnet.name, ])
+        writer.writerow([subnet.subnet, ])
+        writer.writerow('')
+        fields = [IpAddress._meta.get_field('ip_address'), IpAddress._meta.get_field('description')]
+        writer.writerow(field.name for field in fields)
+        for obj in queryset:
+            row = []
+            for field in fields:
+                row.append(str(getattr(obj, field.name)))
+            writer.writerow(row)
+
+    def import_csv(self, file):
+        IpAddress = swapper.load_model("django_ipam", "IpAddress")
+        Subnet = swapper.load_model("django_ipam", "Subnet")
+        reader = csv.reader(StringIO(file.read().decode('utf-8')), delimiter=',')
+        subnet = Subnet.objects.get_or_create(name=next(reader)[0].strip(), subnet=next(reader)[0].strip())[0]
+        next(reader)
+        next(reader)
+        for row in reader:
+            IpAddress.objects.get_or_create(subnet=subnet,
+                                            ip_address=row[0].strip(),
+                                            description=row[1].strip())
```

Comment:
same here, you should do something like @rohithasrk has done in django-freeradius:

- check if ip exists, if does, continue
- create model instances and call `full_clean` but do not call `save` yet
- if validation fails, abort the operation
- otherwise put the model instance in a list
- once all instances have been validated, iterate the list again to save the instances